### ISSUE TITLE
Update Select to include an accessible label as part of the widget

### ISSUE THREAD

**Issue by leapingfrogs** [NoRedInk/noredink-ui#680]
Some analysis shows current select box usage as:

| Usage | Count |
| --- | --- |
| No label or accessibility | 18% |
| Visible accessible label | 20% |
| Hidden accessible label | 44% |
| Custom HTML tag or CSS | 18% |

The default case should give a text label for the select box and allow overriding to hide the label from display while preserving it for screen readers.

**Comment by tesk9** [comment]
When adding a visible label to a select box, note that the label should be "nudged" over from the left, so that it's stylistically consistent with the text input labels.

Please see https://github.com/NoRedInk/NoRedInk/pull/32502#issuecomment-805370784 and the comments after it for context.

**Comment by bendansby** [comment]
Nudged and integrated, actually!

<img width="373" alt="image" src="https://user-images.githubusercontent.com/13528834/130702869-51ae948d-cd8b-4427-95bc-0301050c29dd.png">

https://zpl.io/2ZKWWGV
